ext/test/zpages/tracez_data_aggregator_tests.cc

```diff
@@ -0,0 +1,531 @@
+#include "opentelemetry/sdk/trace/recordable.h"
+#include "opentelemetry/ext/zpages/tracez_data_aggregator.h"
+#include "opentelemetry/ext/zpages/tracez_processor.h"
+#include "opentelemetry/sdk/trace/tracer.h"
+
+#include <gtest/gtest.h>
+
+using namespace opentelemetry::sdk::trace;
+using namespace opentelemetry::ext::zpages;
+namespace nostd  = opentelemetry::nostd;
+namespace common = opentelemetry::common;
+using opentelemetry::v0::trace::Span;
+using opentelemetry::core::SteadyTimestamp;
+
+const std::string span_name1 = "span 1"; 
+const std::string span_name2 = "span 2";
+const std::string span_name3 = "span 3";
+
+/** Test to check if data aggrgator works as expected when there are no spans **/
+TEST(TracezDataAggregator, NoSpans)
+{
+  //Setup
+  std::shared_ptr<TracezSpanProcessor> processor(new TracezSpanProcessor());
+  auto tracer = std::shared_ptr<opentelemetry::trace::Tracer>(new Tracer(processor));
+  auto tracez_data_aggregator (new TracezDataAggregator(processor));
+  
+  const std::map<std::string, std::unique_ptr<AggregatedSpanData>>& data = tracez_data_aggregator->GetAggregatedData();
+  ASSERT_EQ(data.size(), 0); 
+}
+
+/** Test to check if data aggrgator works as expected when there are is exactly one single running span **/
+TEST(TracezDataAggregator, SingleRunningSpan)
+{
+  //Setup
+  std::shared_ptr<TracezSpanProcessor> processor(new TracezSpanProcessor());
+  auto tracer = std::shared_ptr<opentelemetry::trace::Tracer>(new Tracer(processor));
+  auto tracez_data_aggregator (new TracezDataAggregator(processor));
+  
+  //Start the span get the data
+  auto span_first  = tracer->StartSpan(span_name1);
+  const std::map<std::string, std::unique_ptr<AggregatedSpanData>>& data = tracez_data_aggregator->GetAggregatedData();
+  
+  //Check the size and to see if the running span count is accurate
+  ASSERT_EQ(data.size(),1);
```

Comment:
Consider using EXPECT_EQ instead of ASSERT_EQ, similarly EXPECT_TRUE,...

The difference between ASSERT and EXPECT is that when ASSERT fails the test stops. When EXPECT fails the test continues, and eventually reports all the failed expectations.

Reply:
I will look into this, in some cases I do want to use ASSERT because I ASSERT to see if the string name is found in the map and then I look for the name in the map using .at so in this case if the test does not fail on the ASSERT it's going to throw errors during runtime